app/src/utils/routingUtils.ts

```diff
@@ -1,19 +1,40 @@
 export const RETURN_URL_URL_PARAM = "returnUrl";
 
 /**
- * Gets the current url path and query string, excluding the domain and protocol.
+ * Creates a URL to redirect to the login page with the current path as the return URL
  */
-const getCurrentUrlPath = (): string => {
-  return window.location.pathname + window.location.search;
-};
+export function createLoginRedirectUrl() {
+  const path = window.location.pathname;
+  const basename = window.Config.basename;
+  const searchParams = window.location.search;
+  let basePath = "";
+  let pathAfterBase = "";
+  if (basename === "" || basename === "/") {
```

Comment:
can basename be undefined?

Reply:
typed as a string. i can add a check to be safe.